### PR TITLE
CustomSingleChildLayout should support listenable

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/rendering/shifted_box.dart [hunk 1]
```diff
@@ -699,12 +701,37 @@ class RenderFractionallySizedOverflowBox extends RenderAligningShiftedBox {
...
+/// calls [getPositionForChild] to determine the child's position.
+///
+/// The [shouldRelayout] method is called when a new instance of the class
+/// is provided, to check if the new instance actually represents different
+/// information.
+///
+/// The most efficient way to trigger a relayout is to supply a relayout
+/// argument to the constructor of the [SingleChildLayoutDelegate]. The custom
+/// object will listen to this animation and relayout whenever the animation
+/// ticks, avoiding both the build and layout phases of the pipeline.
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/shifted_box.dart]: i doubt it'll relayout in a way that avoids the layout phase...

File: packages/flutter/test/widgets/custom_single_child_layout_test.dart [hunk 2]
```diff
@@ -65,6 +66,25 @@ class FixedSizeLayoutDelegate extends SingleChildLayoutDelegate {
   }
 }
 
+class SizeNotifierLayoutDelegate extends SingleChildLayoutDelegate {
```
Comment (Hixie) [inline packages/flutter/test/widgets/custom_single_child_layout_test.dart]: i'm worried that people will interpret this as "it notifies me when the size changes" as opposed to "i will notify it when the size changes".

Comment (Hixie) [inline packages/flutter/test/widgets/custom_single_child_layout_test.dart]: but this is in a test so whatever

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/rendering/shifted_box.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/rendering/shifted_box.dart]: Oops.  :)

File: packages/flutter/test/widgets/custom_single_child_layout_test.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/test/widgets/custom_single_child_layout_test.dart]: I'll rename it anyway.

**Pull request merged**